### PR TITLE
Add support for rpi2

### PR DESCRIPTION
**Pull request by diederikdehaas**
The kernel upgrade script (https://github.com/debian-pi/raspbian-tools/blob/master/raspbian-kernel-post-install) isn't ready yet, but I think the part of the installer is.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -73,153 +75,186 @@ function create_tempfile {
...
-# extract debs
-for i in packages/*.deb; do
-    cd tmp && ar x ../$i && tar -xf data.tar.*; rm data.tar.*; cd ..
-done
+    # initialize rootfs
+    rm -rf rootfs
+    mkdir -p rootfs
+    # create all the directories needed to copy the various components into place
+    mkdir -p rootfs/bin/
+    mkdir -p rootfs/lib/
```
Comment (goranche) [inline build.sh]: this line can be removed, because the next mkdir will create this dir anyway... (or remove the -p params)

File: build.sh [hunk 2]
```diff
@@ -73,153 +75,186 @@ function create_tempfile {
...
+    mkdir -p rootfs/lib/modules/${KERNEL_VERSION}
+    mkdir -p rootfs/sbin/
+    mkdir -p rootfs/usr/bin/
+    mkdir -p rootfs/usr/share/
```
Comment (goranche) [inline build.sh]: this line can be removed, because the next mkdir will create this dir anyway... (or remove the -p params)

File: build.sh [hunk 3]
```diff
@@ -73,153 +75,186 @@ function create_tempfile {
...
+    cd tmp && ar x ../$i && tar -xf data.tar.*; rm data.tar.*; cd ..
+done
 
-cp installer.cpio bootfs/
 
-echo "kernel=kernel_install.img" > bootfs/config.txt
-echo "initramfs installer.cpio" >> bootfs/config.txt
+# initialize bootfs
+rm -rf bootfs
+mkdir -p bootfs
```
Comment (goranche) [inline build.sh]: the -p parameter is unneeded here

File: scripts/etc/init.d/rcS [hunk 4]
```diff
@@ -99,12 +99,37 @@ tee < ${LOGFILE}.pipe $LOGFILE &
...
+kernel_meta_package=""
+
+if [ "$rpi_hardware" = "$RPI1_HARDWARE" ] ; then
+    rpi1="true"
+    kernel_meta_package="linux-image-rpi-rpfv"
+elif [ "$rpi_hardware" = "$RPI2_HARDWARE" ] ; then
+    rpi2="true"
+    kernel_meta_package="linux-image-rpi2-rpfv"
+else
+    # we're screwd
```
Comment (goranche) [inline scripts/etc/init.d/rcS]: we could still spit out some warning / error / something  :smile:

File: scripts/etc/init.d/rcS [hunk 5]
```diff
@@ -99,12 +99,37 @@ tee < ${LOGFILE}.pipe $LOGFILE &
 exec &> ${LOGFILE}.pipe
 rm ${LOGFILE}.pipe
 
+RPI1_HARDWARE="BCM2708"
+RPI2_HARDWARE="BCM2709"
+rpi_hardware=$(grep Hardware /proc/cpuinfo | cut -d " " -f 2)
+rpi1="false"
```
Comment (goranche) [inline scripts/etc/init.d/rcS]: could a single rpi_version variable do the same? empty => unknown, 1 => 1, 2 => 2...
version checking becomes easier, echoing the RPi version is easy, ...

File: scripts/etc/init.d/rcS [hunk 6]
```diff
@@ -113,34 +138,7 @@ echo -n "Copying boot files... "
...
-    cp /boot/post-install.txt /bootfs/post-install.txt || fail
-fi
-
-if [ -d /boot/config ] ; then
-    cp -r /boot/config /bootfs/ || fail
-fi
-
-# if installation aborts it will restart
-cp /boot/config.txt /bootfs/config.txt || fail
+cp -r /boot/ /bootfs/ || fail
```
Comment (goranche) [inline scripts/etc/init.d/rcS]: this makes me a bit nervous :confused: might this be better? (just added an asterisk)

    cp -r /boot/* /bootfs/ || fail

Comment (diederikdehaas) [inline scripts/etc/init.d/rcS]: It is indeed better. Without the asterisk it creates a boot directory under bootfs.

**Comment by Mausy5043** [comment]
It surprises me to see that you're pulling RPi2 support into the 1.0.x branch in stead of the 1.1.x.

Anyways. I guess it's time for me to re-align my own fork with the current state of the 1.0.x branch and see if I can bother you guys with some new bugs :wink:

**Comment by diederikdehaas** [comment]
The decision is based on whether new parameters need to be introduced or not, see [Semantec Versioning](http://semver.org)

**Comment by goranche** [comment]
> see if I can bother you guys

them'z 'r' fightin' wordz  :grimacing: :wink:

**Pull request merged**